walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,136 @@
+
+        if (event.getAttendees().contains(person)) {
+          if (!eventTitles.contains(event.getTitle())) {
+            allEventTimes.add(event.getWhen());
+            System.out.println("add event: START: " + event.getWhen().start() + " END: " + event.getWhen().end());
+          }
+        }
+      }
+    }
+
+    // no attendees match any events
+    if (allEventTimes.isEmpty()) {
+      System.out.println("no attendees match events");
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    Collections.sort(allEventTimes, TimeRange.ORDER_BY_END); //TO DELETE
+
+
+    System.out.println("sorting complete"); // TO DELETE
+    for (TimeRange aEvent : allEventTimes) {
+      System.out.println("postsort. START: " + aEvent.start() + " END: " + aEvent.end());
+    }
+
+    // Create new time which starts at first opening and ends at  of request
+
+    newTime: 
+    for(TimeRange eventTime : allEventTimes) { 
+      currEnd = eventTime.start();
+      System.out.println("current start: " + currStart + " current end: " + currEnd);
+      TimeRange proposeTime = TimeRange.fromStartEnd(currStart, currEnd, false);
+
+      // check that proposed time is longer than request dureation. Also ensures
+      // that the propsed duration is not negtive
+      if (checkDuration(proposeTime, request.getDuration())) {
+        System.out.println("durration large enough");
+        for (TimeRange anEvent : allEventTimes) {
+          if (anEvent.equals(proposeTime)) {
+            continue;
+          } else if (anEvent.overlaps(proposeTime)) {
+            proposeTime = fixRange(anEvent, proposeTime);
+            if (!checkDuration(proposeTime, request.getDuration())) {
+              currStart = eventTime.end();
+              continue newTime;
+            }
+          }
+        }
+
+        // if the proposed time is already within the collection don't add
+        for (TimeRange RequestOptions : collection) {
+          if (RequestOptions.equals(proposeTime)) {
+            currStart = eventTime.end();
+            continue;
+          }
+        }
+
+        collection.add(proposeTime);
+      }
+      else {
+        System.out.println("duration too short: " + proposeTime.duration());
+      }
+      currStart = eventTime.end();
+    }
+    //  add time gap for end of day, current start is on last 
+    TimeRange proposeTime = TimeRange.fromStartEnd(currStart, DAY_IN_MINS, false);
+    if (proposeTime.duration() > request.getDuration()) {
+      collection.add(proposeTime);
+    }
+
+    return collection;
+    
+  }
+  
+  private boolean checkDuration(TimeRange proposed, long duration) {
+    return proposed.duration() >= duration;
+  }
+
+  private TimeRange fixRange(TimeRange event, TimeRange proposed) {
+    if (event.start() < proposed.start()) {
+      System.out.println("proposed start is within another events start, adusting end");
+      return TimeRange.fromStartEnd(proposed.start(), event.start(), false);
```

Comment:
proposed.start > event.start so doesn't this create a time range with the start > end which seems invalid.